src/types/user.ts

```diff
@@ -0,0 +1,6 @@
+export interface User {
+  id?: number | undefined,
+  name?: string | undefined,
+  username?: string | undefined,
+  email?: string | undefined,
+}
```

Comment:
I don't think that it is a good idea to declare that all parameters are optional
+ we often get data from the format JSON and undefined is not a valid value for JSON